tests/qchem/test_matrices.py

```diff
@@ -655,3 +655,380 @@ def test_core_matrix_diff_positions(self, symbols, geometry, alpha, c_ref):
         args = [mol.coordinates, mol.alpha, r_basis]
         c = qchem.core_matrix(mol.basis_set, mol.nuclear_charges, mol.coordinates)(*args)
         assert np.allclose(c, c_ref)
+
+
+def generate_symbols_geometry_alpha():
+    """Generates symbols, geometry and alpha arrays to be reused for the molecule"""
+    symbols = ["H", "H"]
+    geometry = qml.math.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.0]], like="jax")
+    alpha = qml.math.array(
+        [[3.42525091, 0.62391373, 0.1688554], [3.42525091, 0.62391373, 0.1688554]], like="jax"
+    )
+
+    return symbols, geometry, alpha
+
+
+@pytest.mark.jax
+class TestJax:
+    def test_overlap_matrix_jax(self):
+        r"""Test that overlap_matrix returns the correct matrix when using jax."""
+        s_ref = np.array([[1.0, 0.7965883009074122], [0.7965883009074122, 1.0]])
+        symbols, geometry, alpha = generate_symbols_geometry_alpha()
+
+        mol = qchem.Molecule(symbols, geometry, alpha=alpha)
+        args = [geometry, mol.coeff, alpha]
+        s = qchem.overlap_matrix(mol.basis_set)(*args)
+        assert np.allclose(s, s_ref)
```

Comment:
Why does this use `numpy` and not `jax`?

Reply:
Like the assert statement? It does use jax to build the molecule

Reply:
Changed to use `qml.math.allclose`